src/symbols.js

```diff
@@ -736,6 +736,16 @@ var symbols = {
         },
 
         // AMS Misc
+        "\\#": {
+          font: "ams",
```

Comment:
I should have read https://github.com/Khan/KaTeX/blob/master/CONTRIBUTING.md first :-/ I just picked that font because the other symbols used it as well...